activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -78,6 +78,33 @@ def #{method_name}=(value)           # def includes_values=(value)
 
     alias extensions extending_values
 
+    # Allows preloading of associations on a relation or array of records,
+    # in the same way that #preload does. Mutates the +records+ argument.
+    # For example:
+    #
+    #   users = User.all.to_a
+    #   User.load_associations(users, :posts)
+    #   # SELECT "posts".* FROM "posts" WHERE "posts"."user_id" IN (1, 2, 3)
+    #
+    #   users.each do |user|
+    #     user.posts
+    #   end
+    def load_associations(records, *associations)
+      if records.is_a?(ActiveRecord::Relation)
+        records.preload(associations.flatten)
+      else
+        check_if_array_contains_valid_elements!(records)
+        check_if_method_has_arguments!(:load_associations, associations)
+
+        preloader = build_preloader
+        associations.each do |association|
+          preloader.preload records, association
+        end
+        records
+      end
+    end
+    alias_method :preload_associations, :load_associations
```

Comment:
The method `preload_associations` as already been defined in the latest version  by #31250 so I think that this will cause conflicting behaviour.

In #31250, this method is doing the preloading of associations as part of the loading of the relation, so doesn't take any arguments for the different associations, but does accept the records as the preloading can be done on a subset of the records.

I think that this change would still be good without this alias, so maybe it could be removed? 

Otherwise, maybe these methods could be moved to be class methods on `ActiveRecord::Base` as I am not sure if there is use case for using them on a relation (all the examples in the tests are directly on the  `ActiveRecord::Base` subclasses), but there might be other use cases I am not aware of.

Last option would be to rename the other `preload_associations` method, easiest thing to do would be to prefix with `_` as it is not really part of the public API.

Reply:
Thanks for pointing this out! I think it'd be nice to make `preload_associations` part of the public API. It doesn't seem like there's a precedent for prefixing private methods with `_` so I'd like to rename the current method to be more specific—maybe `preload_associations_on_records`?

In terms of the method being defined on Base vs Relation, I wanted to keep the logic close to the original `preload` query method which is defined on Relation.